cloud-sql-proxy/cloud-sql-proxy.sh

```
@@ -364,92 +420,124 @@ function run_validation() {
     if ! timeout 60s beeline -u "${hiveserver_uri}" -e 'SHOW TABLES;' >&/dev/null; then
       err 'Failed to bring up Cloud SQL Metastore'
     else
-      echo 'Cloud SQL Hive Metastore initialization succeeded' >&2
+      log 'Cloud SQL Hive Metastore initialization succeeded'
     fi
 
     # Execute the Hive "reload function" DDL to reflect permanent functions
     # that have already been created in the HiveServer.
     beeline -u "${hiveserver_uri}" -e "reload function;"
-    echo "Reloaded permanent functions"
+    log 'Reloaded permanent functions'
   fi
-   echo 'Validated' >&2
+   log 'Validated Hive functioning'
 }
 
 function install_mysql_cli() {
   if command -v mysql >/dev/null; then
-    echo "MySQL CLI is already installed" >&2
+    log "MySQL CLI is already installed"
     return
   fi
 
-  echo "Installing MySQL CLI ..." >&2
+  log "Installing MySQL CLI ..."
   if command -v apt >/dev/null; then
     apt update && apt install default-mysql-client -y
   elif command -v yum >/dev/null; then
     yum -y update && yum -y install mysql
   fi
-  echo "MySQL CLI installed" >&2
+  log "MySQL CLI installed"
+}
+
+function install_postgres_cli() {
+  if command -v psql >/dev/null; then
+    log "POSTGRES CLI is already installed"
+    return
+  fi
+
+  log "Installing POSTGRES CLI ..."
+  if command -v apt >/dev/null; then
+    apt update && apt install postgresql-client -y
+  elif command -v yum >/dev/null; then
+    yum -y update && yum -y install postgresql
+  fi
+  echo "POSTGRES CLI installed" >&2
+}
+
+function install_db_cli() {
+  case ${METASTORE_INSTANCE_TYPE} in
+    MYSQL)
+      install_mysql_cli
+      ;;
+    POSTGRES)
+      install_postgres_cli
+      ;;
+    SQL)
+      # TODO: add SQL support
```

If SQLServer is not supported yet, consider fail fast.